src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,106 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public IEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties).GetAwaiter().GetResult();
+
+            int tableCompletionCount = 0;
+
+            using (var frames = pDataSet.GetFrames())
+            {
+                while (frames.MoveNext())
+                {
+                    var frame = frames.Current;
+
+                    switch (frame.FrameType)
+                    {
+                        case FrameType.TableFragment:
+                            {
+                                var content = frame as ProgressiveDataSetDataTableFragmentFrame;
+                                object[] row;
+                                while(GetNextRow(content, out row))
+                                {
+                                    yield return row;
+                                };
+                            }
+                            break;
+
+                        case FrameType.DataTable:
+                            {
+                                var content = frame as DataTable;
+                                foreach (DataRow dataTableRow in content.Rows)
+                                {
+                                    yield return dataTableRow.ItemArray;
+                                }
+                            }
+                            break;
+
+                        case FrameType.TableCompletion:
+                            tableCompletionCount++;
+                            break;
+
+                        case FrameType.DataSetHeader:
+                        case FrameType.TableHeader:
+                        case FrameType.TableProgress:
+                        case FrameType.DataSetCompletion:
+                        case FrameType.LastInvalid:
+                        default:
+                            break;
+                    }
+
+                    if (tableCompletionCount > 1)
+                    {
+                        throw new ArgumentException("This method does not support multiple data result sets.");
+                    }
+                }
+            }
+
+            bool GetNextRow(ProgressiveDataSetDataTableFragmentFrame frame, out object[] row)
```

Comment:
```suggestion
            private bool GetNextRow(ProgressiveDataSetDataTableFragmentFrame frame, out object[] row)
```

Reply:
Since this method is scoped in another method, is the private keyword necessary? (VS is angry at me when I include it)

Reply:
Bleh. Miscounted curlies. I'm like... 0 for 2 on this PR... aren't I.